### DISCUSSION TITLE
How to set up build for Firebeetle

### DISCUSSION THREAD

**Discussion by blacknell** [arendst/Tasmota#19710] [Q&A] [answered]
I've experimented with limited success at defining a build set up for Firebeetle ESP32. But I get crashes/resets and an intermittent manner that makes it very frustrating - the device just continually reset and rebooted.

FWIW I'm not even sure there are particular advantages to building to such a specific target vs doing so with the standard ESP32 flags.

In `platformio_tasmota_cenv.ini` I've added this section below.

I found that when I remove the comments below I needed the `board_build.partitions` value for it to successfully upload. I got these from what looked like a similar issue on another user's non Firebeetle board.

I see that the `firebeetle32` board target is listed in PlatformIO devices but don't think this means Tasmota does?

Does anyone have any guidance?

```
[env:tasmota32-weatherstation]
extends = env:tasmota32_base
build_flags             = ${env:tasmota32_base.build_flags} -DFIRMWARE_WEATHERSTATION
; platform = espressif32
; board = firebeetle32
; board_build.partitions  = partitions/esp32_partition_app1856k_fs1344k.csv
```

**Comment by sfromis** [comment]
What are you even trying to do differently, instead of a standard precompiled build? What you may have of targets in PlatformIO is usually not relevant to Tasmota builds.

**Reply by blacknell** [reply]
I'm not sure!

I do need custom compiled because I'm enabling less used sensors such as MAX17043 and HRG15 (for example).

I assumed (probably wrongly) that some settings from the board might be leveraged.

**Reply by sfromis** [reply]
Well, then I'd suggest a bog standard build, and adding those drivers. Things like gpio configuration are not inherited from elsewhere, but typically done after having started Tasmota. What you do need to be sure of to build is what line of ESP32 chips it is (original, C3, S2, S3 etc), `tasmota32_base` is for the original, if it uses octal SPI for flash/PSRAM (usually not, and not for original ESP32), or if PSRAM (if present at all) needs non-standard pins, other than gpio 16/17. Probably easiest if you first install precompiled Tasmota, and then upgrade to the result for your custom build.

**Comment by Jason2866** [comment, marked as answer]
And do not mess around with Partition Schemes. It is not needed. Tasmota32 uses by default the safeboot scheme and increases at first start the Filesystem to max size by detecting the real flash size.
The APP partition is huge and can be increased (when needed) without the need of reflashing.
We worked hard to provide a few precompiled firmware variants which do use all resources of the device without specifying this details! AFAIK there is no other project which is able to do this.
So long story short. No need for any special boards definitions.

**Reply by blacknell** [reply]
Ok. Thanks both of you. That's been very helpful.

**Comment by blacknell** [comment]
Probably off topic but how to I set up Tasmota so the template web form has only the pins for the Firebeetle.

It defaults to generic dev kit (ie all pins) but I have many Firebeetles and it would be helpful to limit this list to those on that board.

It's probably related to the Based On dropdown and this has only one option, and looking through the code I can't work out the right way to change

**Reply by sfromis** [reply]
Not "Based on", but what you can do is to make a template where unavailable pins are flagged as "None", while open pins are "User". Then you could copy/paste that template to more fire beetles, via "Configure Other". In each device, you can then configure specific pins in "Configure Module", or using "User" pins in Configure Template. A strong point of using templates is that it is easy to store that string, send it, or whatever.

If you get to a good generic template matching the hardware in Firebeetle, you could submit it to https://templates.blakadder.com/ for others to enjoy.

**Comment by blacknell** [comment]
Ok. Will do

I guess the None pins still show up in the template UI which isn’t perfect but will have to do.

thanks

I just wondered why/how on some templates the pin list is restricted. 

Regards
Paul

**Reply by sfromis** [reply]
You may be mixing up "Configure Template" showing all pins, and "Configure Module" only showing "User" pins which are not "taken" in the template - or included in a preconfigured module which is generally not used for ESP32.
